packages/trpc/server/routers/viewer/availability.tsx

```diff
@@ -157,15 +92,51 @@ export const availabilityRouter = router({
             code: "UNAUTHORIZED",
           });
         }
-        const availability = convertScheduleToAvailability(schedule);
         return {
-          schedule,
+          id: schedule.id,
+          name: schedule.name,
           workingHours: getWorkingHours(
             { timeZone: schedule.timeZone || undefined },
             schedule.availability || []
           ),
-          availability,
+          schedule: schedule.availability,
+          availability: convertScheduleToAvailability(schedule).map((a) =>
+            a.map((startAndEnd) => ({
+              ...startAndEnd,
+              // Turn our limited granularity into proper end of day.
+              end: new Date(startAndEnd.end.toISOString().replace("23:59:00.000Z", "23:59:59.999Z")),
+            }))
+          ),
           timeZone: schedule.timeZone || user.timeZone,
+          dateOverrides: schedule.availability.reduce((acc, override) => {
+            // only iff future date override
+            if (!override.date || override.date < new Date()) {
```

Comment:
I can save date overrides for march 1st and 2nd (my date/time saving was march 1st at 11PM GMT-5). It is saved successfully in the database however the date overrides don't show up in the list. 

That's why:
- `new Date()` is utc so it is already march 2nd
- `override.date` has always timestamp 00:00 so it is also smaller than the utc time which was in my case 2023-03-02T04:00

Reply:
Nice find this bug! Fixed it now ⛅